drivers/mpsl/clock_control/clock_control_mpsl.c

```diff
@@ -207,9 +240,45 @@ static enum clock_control_status clock_get_status(struct device *dev,
 	}
 }
 
+static int clock_async_start(struct device *dev, clock_control_subsys_t subsys,
+			     struct clock_control_async_data *data)
+{
+	struct mpsl_clock_control_data *mpsl_control_data = dev->driver_data;
+	enum clock_control_status clock_status = clock_get_status(dev, subsys);
+	sys_slist_t *list = &(mpsl_control_data->async_on_list);
+
+	switch (clock_status) {
+	case CLOCK_CONTROL_STATUS_ON:
+		/* LFCLK is started by default and handled as STATUS_ON */
+		if ((data != NULL) && (data->cb != NULL)) {
+			data->cb(dev, data->user_data);
+		}
+
+		return 0;
+	case CLOCK_CONTROL_STATUS_STARTING:
+	case CLOCK_CONTROL_STATUS_OFF:
+		/* Noop */
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	if ((data != NULL) && (data->cb != NULL)) {
+		/* if same request already scheduled and not yet completed. */
+		if (clock_list_find(list, &data->node)) {
+			return -EBUSY;
+		}
+
+		clock_list_append(list, &data->node);
+	}
+
+	return hf_clock_start(hf_clock_started_callback);
```

Comment:
please note that clock_async_start() will come from LF or HF. Even though LF is in theory always on, devices which requires low frequency clock (like RTC counter) will request it. It should not result in starting hf clock because that will leave HF clock on all the time.

Reply:
In general, LFCLK->clock will be always handled in case 'CLOCK_CONTROL_STATUS_ON' (line 251). Indeed, it is easy to confuse users.

Reply:
To resolve the confusion, let's call clock_start instead.